queue/queue.go

```diff
@@ -0,0 +1,128 @@
+package queue
+
+import (
+	"context"
+	"encoding/binary"
+	"time"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/store"
+)
+
+// Put queues the message in the database to be executed at given time.
+// Due to the implementation details, message is guaranteed to be executed
+// after given time, but not exactly at given time.
+func Put(db weave.KVStore, runAt time.Time, msg weave.Msg) error {
+	msgb, err := msg.Marshal()
+	if err != nil {
+		return errors.Wrap(err, "marshal message")
+	}
+
+	for {
+		key := queueKey(runAt)
+		if ok, err := db.Has(key); err != nil {
+			return errors.Wrap(err, "cannot check key existance")
+		} else if ok {
+			// If the key is already in use, instead of storing a
```

Comment:
:+1:

Fair tradeoff. We need to ensure this queue cannot be spammed in any case